heron_wsgi/admin_lib/i2b2metadata.py

```diff
@@ -118,16 +119,30 @@ def insert_for(i2b2meta_schema, pid, schema, rc_pids, cols):
     return sql, params
 
 
-def schema_for(i2b2_pid):
+def flipflop_suffix(i2b2meta_schema):
+    '''Figure flip-flop context from externally provided i2b2meta_schema.
+
+    >>> flipflop_suffix("bhmetadataB2")
+    'b2'
+
+    '''
+
+    suffix = i2b2meta_schema[-2:].lower()
+    assert suffix in ("a1", "b2")
+    return suffix
+
+
+def schema_for(i2b2_pid, i2b2meta_schema):
     '''Build schema name from specially formatted HERON project ID.
 
     See also create_redcap_projects task in heron_build.py
 
-    >>> schema_for("REDCap_24")
-    ('24', 'REDCAPMETADATA24')
+    >>> schema_for("REDCap_24", "bhmetadataB2")
+    ('24', 'REDCAPMETADATA24b2')

```

Comment:
Oracle schema names are case-sensitive. They're case-folded to upper-case in many contexts, but we shouldn't assume that here. This should return `REDCAPMETADATA24B2`.